### PR TITLE
Adds unit test for result worker (EXPOSUREAPP-4663)

### PR DESCRIPTION
**Pull request by chiljamgossow**
Adds unit test for DiagnosisTestResultRetrievalPeriodicWorker
Removes instrumentation test

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt [hunk 1]
```diff
@@ -54,7 +48,7 @@ class DiagnosisTestResultRetrievalPeriodicWorker @AssistedInject constructor(
         var result = Result.success()
         try {
 
-            if (abortConditionsMet()) {
+            if (abortConditionsMet(System.currentTimeMillis())) {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt:51]: We could inject `TimeStamper` into the worker as use it as time source, then we have full control over the time.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt [hunk 2]
```diff
@@ -84,7 +78,8 @@ class DiagnosisTestResultRetrievalPeriodicWorker @AssistedInject constructor(
         return result
     }
 
-    private fun abortConditionsMet(): Boolean {
+    @VisibleForTesting
+    fun abortConditionsMet(currentMillis: Long): Boolean {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt:82]: If we mock time using `TimeStamper`, we might not need `VisibleForTesting`.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt [hunk 3]
```diff
@@ -75,7 +69,7 @@ class DiagnosisTestResultRetrievalPeriodicWorker @AssistedInject constructor(
                     stopWorker()
                 }
             }
-        } catch (e: Exception) {
+        } catch (ex: Exception) {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt:72]: Can we add a test for the error case, or did I miss it?

**Review by d4rken — COMMENTED** [review]
Looks good :coffee: , a few small tweaks to discuss, but no blockers.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt:51]: Yes, will do

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt]: done

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt (hunk 3 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt:73]: done

**Review by d4rken — APPROVED** [review]
lgtm :+1:

**Comment by d4rken** [comment]
There is currently still an issue with our CI instrumentation tests, if only that check fails, it can still be merged. Will be fixed in #2222

**Review by ralfgehrer — APPROVED** [review]

**Pull request merged**